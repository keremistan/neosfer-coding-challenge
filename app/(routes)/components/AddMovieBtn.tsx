'use client';

import { useState } from "react";
import { BsPlusSquareFill } from "react-icons/bs";

export const AddMovieBtn = (
    { onAddMovie }: { onAddMovie: () => void }
) => {

    const [movieName, setMovieName] = useState<string>("");

    const handleClick = async () => {
        let apiRes = await fetch("/api/addMovie", {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({ title: movieName })
        });

        console.log(await apiRes.json());

        onAddMovie();
    }

    return (
        <div>
            <button className="btn" onClick={() => {
                const modal = document.getElementById('add_movie_modal_1');
                if (modal) {
                    (modal as HTMLDialogElement).showModal();
                }
            }}><BsPlusSquareFill size={24} color="gray" /> Add Movie</button>

            <dialog id="add_movie_modal_1" className="modal">
                <div className="modal-box">
                    <h3 className="font-bold text-lg">Add a new movie</h3>
                    <div className="modal-action justify-center">
                        <form method="dialog">

                            <input
                                type="text"
                                placeholder="Movie name"
                                className="input input-bordered w-full mb-6"
                                onChange={(e) => setMovieName(e.target.value)}
                            />

                            {/* if there is a button in form, it will close the modal */}
                            <button className="btn font-light">Close</button>
                            <button className="btn ml-8" onClick={handleClick}>Add</button>
                        </form>
                    </div>
                </div>
            </dialog>
        </div>
    );
}