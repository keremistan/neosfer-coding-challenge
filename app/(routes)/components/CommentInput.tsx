'use client';

import { useState } from "react";
import { FaTrash, FaArrowRight } from "react-icons/fa";

type CommentInputProps = {
    movieId: number;
    movieComment: string | null;
    onCommentChange: () => void;
};

export const CommentInput = ({ movieId, movieComment, onCommentChange }: CommentInputProps) => {
    const [comment, setComment] = useState("");

    const handleCommentChange = async () => {
        await fetch(`/api/updateComment/${movieId}`, {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({ comment })
        });

        onCommentChange();

        setComment("");
    };

    const handleRemoveComment = async () => {
        await fetch(`/api/updateComment/${movieId}`, {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({ comment: "" })
        });

        onCommentChange();
    }

    if (movieComment) {
        return (
            <div className="flex">
                <p>{movieComment}</p>
                <button className="ml-4" onClick={handleRemoveComment} aria-label="Remove Comment">
                    <FaTrash />
                </button>
            </div>
        );
    }

    return (
        <div className="border-2 rounded flex">
            <input
                type="text"
                value={comment}
                onChange={(e) => setComment(e.target.value)}
                className="p-2 w-full"
                // placeholder="add comment"
                
            />
            <button className="mx-4" onClick={handleCommentChange} aria-label="Submit Comment">
                <FaArrowRight />
            </button>
        </div>
    );
};
