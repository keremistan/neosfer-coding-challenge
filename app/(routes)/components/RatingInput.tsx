'use client';
import React from "react";
import { FaStar } from "react-icons/fa";

type RatingInputProps = {
    movieId: number;
    movieRating: number | null;
    onRatingChange: () => void;
};

export const RatingInput = ({ movieId, movieRating, onRatingChange }: RatingInputProps) => {

    const handleRatingChange = async (newRating: number) => {
        await fetch(`/api/updateRating/${movieId}`, {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({ rating: newRating })
        });

        onRatingChange();
    };

    return (
        <div>
            <div className="flex justify-end items-center">
                <FaStar color="orange" />

                <select className="select select-bordered" onChange={(e) => handleRatingChange(parseInt(e.target.value))}>
                    <option disabled selected={movieRating == null}>R</option>
                    <option selected={movieRating == 1} value={1}>1</option>
                    <option selected={movieRating == 2} value={2}>2</option>
                    <option selected={movieRating == 3} value={3}>3</option>
                    <option selected={movieRating == 4} value={4}>4</option>
                    <option selected={movieRating == 5} value={5}>5</option>
                </select>

            </div>
        </div>
    );
};