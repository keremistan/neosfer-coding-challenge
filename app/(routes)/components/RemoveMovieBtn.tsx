'use client';

import { FaTrash } from "react-icons/fa";

export const RemoveMovieBtn = (
    { movieId, movieTitle, onRemoveMovie }: { movieId: number, movieTitle: string; onRemoveMovie: () => void }
) => {

    const handleClick = () => {
        let apiRes = fetch("/api/removeMovie", {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({ id: movieId })
        });

        apiRes.then((res) => res.json()).then((data) => console.log(data));

        onRemoveMovie();
    }

    return (<div>
        <button className="btn font-light btn-ghost" onClick={() => {
            const modal = document.getElementById("remove_movie_modal_" + movieId);
            if (modal) {
                (modal as HTMLDialogElement).showModal();
            }
        }}>
            <FaTrash size={16} color="gray" />
        </button>


        <dialog id={"remove_movie_modal_" + movieId} className="modal">
            <div className="modal-box">
                <h3 className="font-bold text-lg">Delete movie</h3>
                <div className="modal-action">
                    <form method="dialog">

                        <p className="mb-4">Do you really want to remove the following movie from your favourite movie library?</p>
                        <p className="mb-8 text-2xl">{movieTitle}</p>

                        {/* if there is a button in form, it will close the modal */}
                        <button className="btn font-light">Cancel</button>
                        <button className="btn ml-8 btn-error" onClick={handleClick}>Delete</button>
                    </form>
                </div>
            </div>
        </dialog>
    </div>);
}