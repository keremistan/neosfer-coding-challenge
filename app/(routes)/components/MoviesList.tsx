'use client';

import { useState, useEffect } from "react";
import { AddMovieBtn } from "../add_movie_btn";
import { RemoveMovieBtn } from "../remove_movie_btn";
import { CommentInput } from "./CommentInput";
import { RatingInput } from "./RatingInput";

type Movie = {
    id: number;
    title: string;
    comment: string | null;
    rating: number | null;
}

export const MoviesList = () => {
    const [movies, setMovies] = useState<Movie[]>([]);
    const [searchQuery, setSearchQuery] = useState<string>("");

    const fetchMovies = async (query: string = "") => {
        const response = await fetch(`/api/getAllMovies?query=${query}`);
        const data = await response.json();
        console.log("Movies fetched!", data);
        setMovies(data);
    };

    useEffect(() => {
        fetchMovies();
    }, []);

    useEffect(() => {
        fetchMovies(searchQuery);
    }, [searchQuery]);

    return (
        <div>
            <h1>Movies</h1>
            <input
                type="text"
                placeholder="Search movies..."
                value={searchQuery}
                onChange={(e) => setSearchQuery(e.target.value)}
            />
            <ul>
                {movies.map((movie) => (
                    <li key={movie.id}>
                        <h2 className="text-3xl text-tahiti">{movie.title}</h2>
                        <p className="text-red font-medium">Comment: {movie.comment == undefined ? "N/A" : movie.comment.length == 0 ? "No Comment" : movie.comment}</p>
                        <CommentInput movieId={movie.id} onCommentChange={() => fetchMovies(searchQuery)} />
                        <p>Rating: {movie.rating || "N/A"}</p>
                        <RatingInput movieId={movie.id} onRatingChange={() => fetchMovies(searchQuery)} />
                    </li>
                ))}
            </ul>
            <AddMovieBtn />
            <RemoveMovieBtn />
        </div>
    );
};