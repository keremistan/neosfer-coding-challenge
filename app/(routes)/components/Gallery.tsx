'use client';

import React, { createContext } from "react";
import { useState, useEffect } from "react";
import { AddMovieBtn } from "./AddMovieBtn";
import { MovieType } from "../../types/movie";
import { MovieCard } from "./MovieCard";

type HomeSearchQueryContextType = {
    searchQuery: string;
};

export const HomeSearchQueryContext = createContext<HomeSearchQueryContextType>({ searchQuery: '' });

export const Gallery = () => {
    const [movies, setMovies] = useState<MovieType[]>([]);
    const [searchQuery, setSearchQuery] = useState<string>("");

    const fetchMovies = async (query: string = "") => {
        const response = await fetch(`/api/getAllMovies?query=${query}`);
        const data = await response.json();
        console.log("Movies fetched!", data);
        setMovies(data);
    };

    useEffect(() => {
        fetchMovies();
    }, []);

    useEffect(() => {
        fetchMovies(searchQuery);
    }, [searchQuery]);

    return (
        <HomeSearchQueryContext.Provider value={{ searchQuery }}>
            <div className="m-4">
                {/* Title and adding a movie button */}
                <div className="flex justify-between items-center mb-4">
                    <h1 className="text-2xl font-bold">MovieLibs</h1>
                    <AddMovieBtn onAddMovie={() => fetchMovies()} />
                </div>

                {/* Search field */}
                <input
                    className="input input-bordered w-full mb-4"
                    type="text"
                    placeholder="Search movies..."
                    value={searchQuery}
                    onChange={(e) => setSearchQuery(e.target.value)}
                />

                {/* Laying out the movie cards */}
                <div className="md:flex md:flex-wrap">
                    {movies.map((movie) => (
                        <MovieCard key={movie.id} movie={movie} fetchMovies={fetchMovies} />
                    ))}
                </div>
            </div>
        </HomeSearchQueryContext.Provider>
    );
};