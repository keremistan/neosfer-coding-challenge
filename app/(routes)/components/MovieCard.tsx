import React, { useContext } from 'react';
import { CommentInput } from './CommentInput';
import { RatingInput } from './RatingInput';
import { RemoveMovieBtn } from './RemoveMovieBtn';
import { FaBookmark, FaRegBookmark } from 'react-icons/fa';
import { MovieType } from '../../types/movie';
import { HomeSearchQueryContext } from './Gallery';
import { sendApiRequest } from '@/app/utility/sendApiRequest';

type MovieCardProps = {
    movie: MovieType;
    fetchMovies: (searchQuery: string) => void;
};

export const MovieCard = ({ movie, fetchMovies }: MovieCardProps) => {

    const { searchQuery } = useContext(HomeSearchQueryContext);

    const handleWishlistToggle = async (movieId: number) => {
        let toggleRes = await sendApiRequest(`/api/toggleWishlist?movieId=${movieId}`, "GET");

        console.log({ toggleRes });

        fetchMovies(searchQuery);
    }

    return (
        <div className="container p-4 rounded w-full md:w-1/3" key={movie.id}>
            <div className="bg-white p-4 rounded-lg drop-shadow hover:drop-shadow-lg">

                {/* Card title and movie delete button */}
                <div className="group flex justify-between items-center">
                    <h2 className="text-2xl mb-2 text-gray-500">{movie.title}</h2>
                    <div className="hidden group-hover:block">
                        <RemoveMovieBtn key={movie.id} movieId={movie.id} movieTitle={movie.title} onRemoveMovie={() => fetchMovies(searchQuery)} />
                    </div>
                </div>

                {/* Comment input */}
                <div className="my-4">
                    <CommentInput movieId={movie.id} movieComment={movie.comment} onCommentChange={() => fetchMovies(searchQuery)} />
                </div>

                {/* Rating and wishlisting an individual movie */}
                <div className="flex justify-between items-center">
                    <div className="my-4">
                        <RatingInput key={movie.id} movieId={movie.id} movieRating={movie.rating} onRatingChange={() => fetchMovies(searchQuery)} />
                    </div>
                    <div>
                        <button onClick={() => handleWishlistToggle(movie.id)}>
                            {movie.isInWishlist ? <FaBookmark color="gray" /> : <FaRegBookmark color="gray" />}
                        </button>
                    </div>
                </div>
            </div>
        </div>
    );
}