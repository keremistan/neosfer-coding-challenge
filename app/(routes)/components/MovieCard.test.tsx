import { expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import { it } from 'node:test'
import { MovieCard } from '@/app/(routes)/components/MovieCard'


test('MovieCard', () => {

    it('should render and match snapshot', () => {
        const res = render(<MovieCard movie={{ id: 1, title: 'Test Movie', comment: '', rating: 0, isInWishlist: false }} fetchMovies={() => { }} />)
        expect(res).toMatchSnapshot();
    })

})
