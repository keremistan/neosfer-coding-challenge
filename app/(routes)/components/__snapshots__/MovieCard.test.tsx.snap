// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieCard 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container p-4 rounded w-full md:w-1/3"
      >
        <div
          class="bg-white p-4 rounded-lg drop-shadow hover:drop-shadow-lg"
        >
          <div
            class="group flex justify-between items-center"
          >
            <h2
              class="text-2xl mb-2 text-gray-500"
            >
              Test Movie
            </h2>
            <div
              class="hidden group-hover:block"
            >
              <div>
                <button>
                  <svg
                    color="lightgray"
                    fill="currentColor"
                    height="16"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: lightgray;"
                    viewBox="0 0 448 512"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                    />
                  </svg>
                </button>
                <dialog
                  class="modal"
                  id="remove_movie_modal_1"
                >
                  <div
                    class="modal-box"
                  >
                    <h3
                      class="font-bold text-lg"
                    >
                      Delete movie
                    </h3>
                    <div
                      class="modal-action"
                    >
                      <form
                        method="dialog"
                      >
                        <p
                          class="mb-4"
                        >
                          Do you really want to remove the following movie from your favourite movie library?
                        </p>
                        <p
                          class="mb-8 text-2xl"
                        >
                          Test Movie
                        </p>
                        <button
                          class="btn font-light"
                        >
                          Cancel
                        </button>
                        <button
                          class="btn ml-8 btn-error"
                        >
                          Delete
                        </button>
                      </form>
                    </div>
                  </div>
                </dialog>
              </div>
            </div>
          </div>
          <div
            class="my-4"
          >
            <div
              class="rounded flex"
            >
              <input
                class="w-full "
                placeholder="add comment"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="flex justify-between items-center"
          >
            <div
              class="my-4"
            >
              <div>
                <div
                  class="flex justify-end items-center text-gray-500"
                >
                  <svg
                    color="orange"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: orange;"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    />
                  </svg>
                  <select>
                    <option
                      disabled=""
                      value="R"
                    >
                      R
                    </option>
                    <option
                      value="1"
                    >
                      1
                    </option>
                    <option
                      value="2"
                    >
                      2
                    </option>
                    <option
                      value="3"
                    >
                      3
                    </option>
                    <option
                      value="4"
                    >
                      4
                    </option>
                    <option
                      value="5"
                    >
                      5
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <div>
              <button>
                <svg
                  color="gray"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: gray;"
                  viewBox="0 0 384 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container p-4 rounded w-full md:w-1/3"
    >
      <div
        class="bg-white p-4 rounded-lg drop-shadow hover:drop-shadow-lg"
      >
        <div
          class="group flex justify-between items-center"
        >
          <h2
            class="text-2xl mb-2 text-gray-500"
          >
            Test Movie
          </h2>
          <div
            class="hidden group-hover:block"
          >
            <div>
              <button>
                <svg
                  color="lightgray"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: lightgray;"
                  viewBox="0 0 448 512"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </button>
              <dialog
                class="modal"
                id="remove_movie_modal_1"
              >
                <div
                  class="modal-box"
                >
                  <h3
                    class="font-bold text-lg"
                  >
                    Delete movie
                  </h3>
                  <div
                    class="modal-action"
                  >
                    <form
                      method="dialog"
                    >
                      <p
                        class="mb-4"
                      >
                        Do you really want to remove the following movie from your favourite movie library?
                      </p>
                      <p
                        class="mb-8 text-2xl"
                      >
                        Test Movie
                      </p>
                      <button
                        class="btn font-light"
                      >
                        Cancel
                      </button>
                      <button
                        class="btn ml-8 btn-error"
                      >
                        Delete
                      </button>
                    </form>
                  </div>
                </div>
              </dialog>
            </div>
          </div>
        </div>
        <div
          class="my-4"
        >
          <div
            class="rounded flex"
          >
            <input
              class="w-full "
              placeholder="add comment"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="flex justify-between items-center"
        >
          <div
            class="my-4"
          >
            <div>
              <div
                class="flex justify-end items-center text-gray-500"
              >
                <svg
                  color="orange"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: orange;"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <select>
                  <option
                    disabled=""
                    value="R"
                  >
                    R
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div>
            <button>
              <svg
                color="gray"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: gray;"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
